app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java

```diff
@@ -168,7 +168,9 @@ public void queryPairedDevices() {
 
         this.pairedDevices = this.bluetoothAdapter.getBondedDevices();
         for (final BluetoothDevice device : this.pairedDevices) {
-            this.pairedDevicesArrayAdapter.add(device.getName() + "\n" + device.getAddress());
+            if(this.pairedDevicesArrayAdapter.getPosition(device.getName() + "\n" + device.getAddress()) == -1) {
```

Comment:
Can you use String.format? That's a lot faster than string contatenation:
`String.format("%s\n%s", device.getName(), device.getAddress())`